### ISSUE TITLE
Parameterized test fail when test-data is a string containing null-bytes.

### ISSUE THREAD

**Issue by bjornars** [pytest-dev/pytest#2973]
This was introduced by #2734, and released in 3.3 (specifically 78a027e1)

Repro:

```
import base64                              
import pytest                              


def convert(data):                         
    return base64.b16encode(data).lower().decode()                                    


@pytest.mark.parametrize('data,expected', [                                           
    (b'\0\0\0\0', '00000000'),             
])                                         
def test_convert(data, expected):          
    assert convert(data) == expected  
```

Error:
```
self = environ({'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/user/1000/bus'})            
key = b'PYTEST_CURRENT_TEST', value = b'test_repro.py::test_convert[\x00\x00\x00\x00-00000000] (setup)'                                                                      

    def __setitem__(self, key, value):     
        key = self.encodekey(key)          
        value = self.encodevalue(value)    
>       self.putenv(key, value)            
E       ValueError: embedded null byte     

../virtenv/test-venv/lib/python3.5/os.py:731: ValueError                                   

```

An educated guess would be that the parameterized arguments end up in the os.environ in some way, and that fails when the string contains embedded null-bytes. This was exposed with the above mentioned commit changing how strings were escaped and cleaned.

**Comment by The-Compiler** [comment]
Duplicate of #2957

**Comment by The-Compiler** [comment]
Note that a fix is already available in the master branch: #2969
